src/redux/books/booksSlice.js

```diff
@@ -1,46 +1,98 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
 
+const appId = 'q8FdluzHjZBBWXH5Y85r';
 const initialState = {
-  books: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: [],
+  isLoading: false,
+  error: false,
 };
+export const getBooks = createAsyncThunk('books/getBooks', () => axios.get(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${appId}/books`)
+  .then((response) => response.data));
+export const addBook = createAsyncThunk('books/addBook', async (book) => {
+  const result = await axios.post(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${appId}/books`, book);
+  return result.data;
+});
+export const removeBook = createAsyncThunk('books/removeBook', async (id) => {
+  await axios.delete(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${appId}/books/${id}`);
+  return id;
+});
 
 const booksSlice = createSlice({
   name: 'books',
   initialState,
   reducers: {
-    addBook: (state, action) => {
-      state.books.push(action.payload);
-    },
-    removeBook: (state, action) => {
-      const bookId = action.payload;
-      const filteredBooks = state.books.filter((book) => book.item_id !== bookId);
+    addNewBook: (state, action) => {
+      const bookItem = action.payload;
+      const book = {
+        item_id: bookItem.item_id,
+        title: bookItem.title,
+        author: bookItem.author,
+        category: bookItem.category,
+      };
       return {
         ...state,
-        books: filteredBooks,
+        books: [...state.books, book],
       };
     },
-
+  },
+  extraReducers: (builder) => {
+    builder.addCase(getBooks.pending, (state) => ({
+      ...state,
+      isLoading: true,
+      error: false,
+    }));
+    builder.addCase(getBooks.fulfilled, (state, action) => {
+      const books = action.payload;
+      const newBooks = [];
+      Object.keys(books).forEach((book) => newBooks.push({
+        item_id: book,
+        title: books[book][0].title,
+        author: books[book][0].author,
+        category: books[book][0].category,
+      }));
```

Comment:
[OPTIONAL] You did great 👍🏽  but consider using `map` instead of `forEach`: In the `getBooks.fulfilled` action, the code uses `Object.keys` and `forEach` to convert the `books` object into an array. You could use the `map` function instead to simplify the code, like this:
```
const books = action.payload;
const newBooks = Object.keys(books).map((book) => ({
  item_id: book,
  title: books[book][0].title,
  author: books[book][0].author,
  category: books[book][0].category,
}));
```